x-pack/legacy/plugins/siem/public/pages/case/components/configure_cases/index.tsx

```diff
@@ -0,0 +1,205 @@
+import { useCaseConfigure } from '../../../../containers/case/configure/use_configure';
+import {
+  ClosureType,
+  CasesConfigurationMapping,
+  CasesConfigurationMaps,
+} from '../../../../containers/case/configure/types';
+import { Connectors } from '../configure_cases/connectors';
+import { ClosureOptions } from '../configure_cases/closure_options';
+import { Mapping } from '../configure_cases/mapping';
+import { SectionWrapper } from '../wrappers';
+import { configureCasesReducer, State } from './reducer';
+import * as i18n from './translations';
+
+const FormWrapper = styled.div`
+  ${({ theme }) => css`
+    & > * {
+      margin-top 40px;
+    }
+
+    padding-top: ${theme.eui.paddingSizes.l};
+    padding-bottom: ${theme.eui.paddingSizes.l};
+  `}
+`;
+
+const initialState: State = {
+  connectorId: 'none',
+  closureType: 'close-by-user',
+  mapping: null,
+};
+
+const ConfigureCasesComponent: React.FC = () => {
+  const [connectorIsValid, setConnectorIsValid] = useState(true);
+
+  const [{ connectorId, closureType, mapping }, dispatch] = useReducer(
+    configureCasesReducer(),
+    initialState
+  );
+
+  const setConnectorId = useCallback((newConnectorId: string) => {
+    dispatch({
+      type: 'setConnectorId',
+      connectorId: newConnectorId,
+    });
+  }, []);
+
+  const setClosureType = useCallback((newClosureType: ClosureType) => {
+    dispatch({
+      type: 'setClosureType',
+      closureType: newClosureType,
+    });
+  }, []);
+
+  const setMapping = useCallback((newMapping: CasesConfigurationMapping[]) => {
+    dispatch({
+      type: 'setMapping',
+      mapping: newMapping,
+    });
+  }, []);
+
+  const { loading: loadingCaseConfigure, persistLoading, persistCaseConfigure } = useCaseConfigure({
+    setConnectorId,
+    setClosureType,
+  });
+  const {
+    loading: isLoadingConnectors,
+    connectors,
+    refetchConnectors,
+    updateConnector,
+  } = useConnectors();
+
+  const handleSubmit = useCallback(
+    // TO DO give a warning/error to user when field are not mapped so they have chance to do it
+    async () =>
+      // eslint-disable-next-line no-return-await
+      await Promise.all([
+        persistCaseConfigure({ connectorId, closureType }),
+        updateConnector(connectorId, mapping ?? []),
+      ]),
+    [connectorId, closureType, mapping]
+  );
+
+  useEffect(() => {
+    if (
+      !isEmpty(connectors) &&
+      connectorId !== 'none' &&
+      connectors.some(c => c.id === connectorId)
+    ) {
+      const myConnector = connectors.find(c => c.id === connectorId);
+      const myMapping = myConnector?.config?.casesConfiguration?.mapping ?? [];
+      setMapping(
+        myMapping.map((m: CasesConfigurationMaps) => ({
+          source: m.source,
+          target: m.target,
+          actionType: m.action_type,
+        }))
+      );
+    }
+  }, [connectors, connectorId]);
+
+  useEffect(() => {
+    const resetConnectorToDefaultState = async () => {
```

Comment:
why `async`?

Reply:
No reason :). Forgot to remove it. In previous versions, there was a call to an async function inside `resetConnectorToDefaultState `